From df1ebf077f99fef1f56262bbb4c6d75a9405dfd4 Mon Sep 17 00:00:00 2001
From: Martin Sivak <msivak@redhat.com>
Date: Mon, 12 Dec 2011 12:22:42 +0100
Subject: [PATCH] Fix a segfault when overrides line is incomplete

Resolves: rhbz#673100
---
 depmod.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/depmod.c b/depmod.c
index a1d2f8c..d8eccad 100644
--- a/depmod.c
+++ b/depmod.c
@@ -1561,6 +1561,12 @@ static int parse_config_file(const char *filename,
 			version = strsep_skipspace(&ptr, "\t ");
 			subdir = strsep_skipspace(&ptr, "\t ");
 
+                        if (!modname || !version || !subdir) {
+                                warn("syntax error in the override cmd, some values"
+                                     " are missing on line %d\n", linenum);
+                                continue;
+                        }
+
 			if (!regex_match(kernelversion, (const char *)version))
 				continue;
 
-- 
1.7.4.4

