diff -up module-init-tools-3.9/modinfo.c.rhbz882791 module-init-tools-3.9/modinfo.c
--- module-init-tools-3.9/modinfo.c.rhbz882791	2009-05-27 14:33:08.000000000 -0400
+++ module-init-tools-3.9/modinfo.c	2013-06-05 18:08:53.650120214 -0400
@@ -242,7 +242,7 @@ static struct elf_file *grab_module(cons
 			module = grab_elf_file(filename);
 			if (!module)
 				error("modinfo: could not open %s: %s\n",
-					*filename, strerror(errno));
+					filename, strerror(errno));
 			free(filename);
 			return module;
 		}
