diff -uNr module-init-tools-3.9/depmod.c module-init-tools-3.9.new/depmod.c
--- module-init-tools-3.9/depmod.c	2012-02-28 14:59:12.000000000 +0100
+++ module-init-tools-3.9.new/depmod.c	2012-02-28 15:00:22.000000000 +0100
@@ -193,7 +193,7 @@
 {
 	unsigned int dummy;
 
-	return (sscanf(version, "%u.%u.%u", &dummy, &dummy, &dummy) == 3);
+	return (sscanf(version, "%u.%u", &dummy, &dummy) == 2);
 }
 
 static int old_module_version(const char *version)
